pkg/library/mappers/envoysecuritypolicy_to_kuadrant.go

```diff
@@ -0,0 +1,53 @@
+package mappers
+
+import (
+	"context"
+	"fmt"
+
+	egv1alpha1 "github.com/envoyproxy/gateway/api/v1alpha1"
+	kuadrantv1beta1 "github.com/kuadrant/kuadrant-operator/api/v1beta1"
+	"github.com/kuadrant/kuadrant-operator/pkg/library/kuadrant"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type SecurityPolicyToKuadrantEventMapper struct {
+	opts MapperOptions
+}
+
+func NewSecurityPolicyToKuadrantEventMapper(o ...MapperOption) *SecurityPolicyToKuadrantEventMapper {
+	return &SecurityPolicyToKuadrantEventMapper{opts: Apply(o...)}
+}
+
+func (m *SecurityPolicyToKuadrantEventMapper) Map(ctx context.Context, obj client.Object) []reconcile.Request {
```

Comment:
For simplicity, I would just list the kuadrant CRs and return namespaced name of it. 

Other approach, would be to follow the topology. If the secpolicy is targeting a route, get the gateways and one of the gateways should have kuadrant reference in the annotations. If the sec policy is targeting a gateway, the gateway should have the kuadrant CR reference in the annotations.

Reply:
Wouldn't this trigger our controller for updates to SecurityPolicies in the cluster which are not owned by kuadrant? We should at least check that the security policy is managed by us, and if so then it's simple enough to get the namespace of that kuadrant instance?

Reply:
You are right, LGTM about this mapper. We can leverage the annotation `kuadrant.io/namespace`.